ruff.toml

```diff
@@ -0,0 +1,56 @@
+exclude = [
+    ".git",
+    "__pycache__",
+    "setup.py",
+    "build",
+    "dist",
+    "releases",
+    ".venv",
+    ".tox",
+    ".mypy_cache",
+    ".pytest_cache",
+    ".vscode",
+    ".github",
+]
+line-length = 120
+indent-width = 4
+ignore-init-module-imports = true
+
+[lint]
+select = [
+    "E",  # pycodestyle errors
+    "F",  # pyflakes errors
+    "W",  # pycodestyle warnings
+    "C",  # mccabe complexity checker
+    "B",  # bugbear best practices checker
+    "T",  # mypy type errors
+    "B", # flake8-bugbear best practices checker
+    "I001", # isort import order
+]
+ignore = [
+    "E203",  # whitespace before ':'
+    "E722",  # do not use bare except
+    "C401",  # Unnecessary generator (rewrite as a `set` comprehension)
+    "C405",  # Unnecessary list literal - rewrite as a literal
+    "C408",  # Unnecessary dict call - rewrite as a literal
+    "C409",  # Unnecessary list passed to tuple() - rewrite as a tuple literal
+    "C416",  # Unnecessary `set` comprehension (rewrite using `set()`)
+]
```

Comment:
You know what? I would remove them all and add local comments to prevent the linter to break on these.
I don't see why we'd have so many of these as to make these rules global.
My two cents.

Reply:
I added a `TODO` comment for latter